src/main/java/com/epam/processor/DataProcessor.java

```diff
@@ -1,123 +1,233 @@
+						&& accident.getLatitude() <= maxLatitude
+						&& accident.getLatitude() >= minLatitude) {
+					colls.add(accident);
+				}
+			}
+		}
+		return colls;
+	}
+
+	/**
+	 * count incidents by road surface conditions ex: wet -> 2 dry -> 5
+	 * 
+	 * @return
+	 */
+	public Map<String, Long> getCountByRoadSurfaceCondition7() {
+		//roadSurfaceConditions
+		Map<String, Long> resultMap = null;
+		
+		if (roadAccidentList != null && roadAccidentList.size() > 0) {
+			resultMap = new HashMap<String, Long>(); 
+			Multimap<String, RoadAccident> multiMap = ArrayListMultimap.create();
+			for (RoadAccident accident : roadAccidentList) {
+				multiMap.put(accident.getRoadSurfaceConditions(),
+						accident);				
+			}			
+			for(String key:multiMap.keySet()){
+				resultMap.put(key, new Long(multiMap.get(key).size()));
+			}
+		}
+		
+		return resultMap;
+		
+		
+		
+	}
+
+	/**
+	 * find the weather conditions which caused the top 3 number of incidents as
+	 * example if there were 10 accidence in rain, 5 in snow, 6 in sunny and 1
+	 * in foggy, then your result list should contain {rain, sunny, snow} - top
+	 * three in decreasing order
+	 * 
+	 * @return
+	 */
+	public List<String> getTopThreeWeatherCondition7() {
+		List<String> result = null;
+		if (roadAccidentList != null && roadAccidentList.size() > 0) {
+			result = new ArrayList<String>();
+			Map<String, Long> counterMap = new HashMap<String, Long>();			//
+			Multimap<String, RoadAccident> multiMap = ArrayListMultimap.create();
+			for (RoadAccident accident : roadAccidentList) {
+				multiMap.put(accident.getWeatherConditions(),
+						accident);				
+			}
+			
+			for(String key:multiMap.keySet()){
+				counterMap.put(key, new Long(multiMap.get(key).size()));
+			}
+			
+			List<Map.Entry<String, Long>> entryList = new ArrayList<Map.Entry<String, Long>>();
+			
+			for(Entry<String, Long> entry:counterMap.entrySet()){
+				entryList.add(entry);				
+			}
+			Collections.sort(entryList,new Comparator<Map.Entry<String,Long>>(){
+				@Override
+				public int compare(Entry<String, Long> o1,
+						Entry<String, Long> o2) {						
+					return o1.getValue().compareTo(o2.getValue());
+				}
+				
+			});
+			
+			int length = entryList.size();
+			result.add(entryList.get(length-1).getKey());
+			
+			if(entryList.size()>=3){				
+				result.add(entryList.get(length-2).getKey());
+				result.add(entryList.get(length-3).getKey());
+			}else if(entryList.size()>=2){				
+				result.add(entryList.get(length-2).getKey());
+			}
+			
+		}
+		return result;
+	}
+
+	/**
+	 * return a multimap where key is a district authority and values are
+	 * accident ids ex: authority1 -> id1, id2, id3 authority2 -> id4, id5
+	 * 
+	 * @return
+	 */
+	public Multimap<String, String> getAccidentIdsGroupedByAuthority7() {
+		Multimap<String, String> multiMap = null;
+		if (roadAccidentList != null && roadAccidentList.size() > 0) {
+			multiMap = ArrayListMultimap.create(); // ArrayListMultimap allows
```

Comment:
Not sure that I understand your comment correctly. 

Docs for HashMultiMap says that "This multimap does not store duplicate key-value pairs."

So let's say we adding same key-value pair (key1-val1) twice,
Then in HashMultiMap it will be 
key1->val1

when in arraylistMultimap
key1->val1
key1->val1

The question here do you need duplication here in your result? I would use HashMultiMap, but you can come with some usecase to convince me keep duplicates :smile: